specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/examples/PlaybookConfigurations/PutPlaybookConfiguration_example.json

```diff
@@ -0,0 +1,175 @@
+              "scopePath": "/subscriptions/6618f01b-9b61-4cfc-9010-836b34b1bb2a"
+            },
+            {
+              "description": "Some custom description",
+              "scopePath": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23"
+            }
+          ],
+          "sources": [
+            {
+              "eventSource": "AscAlerts",
+              "ruleSets": [
+                {
+                  "rules": [
+                    {
+                      "propertyJPath": "$.type",
+                      "propertyType": "System.String",
+                      "expectedValue": "customType",
+                      "operator": "EqualsIgnoreCase"
+                    }
+                  ]
+                }
+              ]
+            }
+          ],
+          "actions": [
+            {
+              "resourceId": "/subscriptions/e54a4a18-5b94-4f90-9471-bd3decad8a2e/resourceGroups/sample/providers/Microsoft.Logic/workflows/MyTest1",
+              "resourceProviderNamespace": "Microsoft.Logic",
+              "uri": "https://triggerUri1.com"
+            },
+            {
+              "resourceId": "/subscriptions/e54a4a18-5b94-4f90-9471-bd3decad8a2e/resourceGroups/sample/providers/Microsoft.Logic/workflows/MyTest2",
+              "resourceProviderNamespace": "Microsoft.Logic",
+              "uri": "https://triggerUri2.com"
+            }
+          ]
+        }
+      }
+    },
+    "201": {
+      "body": {
+        "id": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23/resourceGroups/myRg/providers/Microsoft.Security/playbookConfigurations/myPlaybookConfigName",
+        "name": "myPlaybookConfigName",
+        "type": "Microsoft.Security/playbookConfigurations",
+        "location": "Central US",
+        "tags": {},
+        "properties": {
+          "description": "Alerts auto dismiss",
+          "isEnabled": true,
+          "metadata": {
+            "createdDateTimeUtc": "2019-03-27T08:43:54.5755094Z",
+            "createdBy": "2c429ed8-089f-42aa-9bb9-92312b8c9f72",
+            "lastUpdatedDateTimeUtc": "2019-03-28T08:43:54.5755094Z",
+            "lastUpdatedBy": "2c429ed8-089f-42aa-9bb9-92312b8c9f72"
+          },
+          "scopes": [
+            {
+              "description": "Some custom description",
+              "scopePath": "/subscriptions/6618f01b-9b61-4cfc-9010-836b34b1bb2a"
+            },
+            {
+              "description": "Some custom description",
+              "scopePath": "/subscriptions/20ff7fc3-e762-44dd-bd96-b71116dcdc23"
+            }
+          ],
+          "sources": [
+            {
+              "eventSource": "AscAlerts",
+              "ruleSets": [
+                {
+                  "rules": [
+                    {
+                      "propertyJPath": "$.type",
+                      "propertyType": "System.String",
+                      "expectedValue": "customType",
+                      "operator": "EqualsIgnoreCase"
```

Comment:
How does this deal with api versions? Resource payloads are allowed to change between API versions, which would invalidate the rules.

Reply:
We have an internal version handling, that also uses the alias mechanism.

Reply:
We are starting with support for JPaths (where versions are handled internally), and we will be adding future support for aliases as well (different scenarios).

Reply:
That's great that you're reusing aliases, but the API suggests that you're operating on the resource body `$.type` looks a lot like querying the "type" property on a resource payload. How does this actually work? Can elaborate on how you handle versions internally?

---
In reply to: [281746903](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r281746903) [](ancestors = 281746903)

Reply:
You are correct, it is a bit confusing. Our choice of example was not a good one. We updated it to reflect a different JPath.

Just to clarify, we are not querying the type at all (as explained, its a bad example on our part). The user specifies a JPath to an object, specifies the type to expect, the value to expect and an operator from a list of known operators (supplied via documentation).

In the future we will support also Aliases instead of specific JPaths.